cmd/root.go

```diff
@@ -37,19 +37,39 @@ func persistentValidateFunc(rootCommand *RootCmd, rootOpts *RootOptions) func(c
 			"output-probe":  "output.probe",
 		}
 		rootCommand.c.Flags().VisitAll(func(f *pflag.Flag) {
-			if name := f.Name; !skip[name] {
-				value := viper.GetString(name)
-				if value == "" {
-					// fallback to nested options in config file, if any
-					if nestedName, ok := nested[name]; ok {
-						value = viper.GetString(nestedName)
-					}
-				}
-				// set the value, if any, otherwise let the default
-				if value != "" {
-					rootCommand.c.Flags().Set(name, value)
-				}
-			}
+		    if name := f.Name; !skip[name] {
+                if name == "kernelurls" {
+                    // Slice types need special treatment when used as flags. If we call 'Set(name, value)',
+                    // rather than replace, it appends. Since viper will already have the cli options set
+                    // if supplied, we only need this step if rootCommand doesn't already have them e.g.
+                    // not set on CLI so read from config.
+                    if cli_urls, err := rootCommand.c.Flags().GetStringSlice(name); err == nil && len(cli_urls) != 0 {
+                       return
+                    }
+                    value := viper.GetStringSlice(name)
+                    if len(value) == 0 {
+                        // Special case for Kernel crawler driverkit output
+                        // TODO, Change kernel crawler to use 'kernelurls' instead of its current 'headers' key
+                        value = viper.GetStringSlice("headers")
```

Comment:
I'd cut this :) i mean, at the moment kernel-crawler output is pre-processed by test-infra makefile (https://github.com/falcosecurity/test-infra/blob/master/driverkit/utils/scrape_and_generate) therefore it is just a matter of updating that file instead of doing this inside driverkit :)

Reply:
Ah nice, have removed in the latest commit.
Also the url formatting will happen in a different ticket. I've been asked to add some RHEL support first as a priority 😅

Reply:
Ah that makes sense :) Thank you; in the end, for now, `kernelurls` is all that is needed given kernel-crawler :)